### DISCUSSION TITLE
list each window's tabs and their titles ??

### DISCUSSION THREAD

**Discussion by GZLiew** [Hammerspoon/hammerspoon#3109] [Q&A] [answered]
Is there a way I can list tabs from each window with each of the tab title

Im trying to find a way where i can fuzzy search all my tabs on all my active windows

**Comment by cmsj** [comment, marked as answer]
There's not really an *easy* way to do that. The ugly way would be to get AppleScript to generate the list for you. The difficult way would be to use `hs.axuielement` to crawl over the UI tree of each window, to find the tab elements.

**Reply by GZLiew** [reply]
@cmsj I guess apple just doesn't give us easy access to certain apis. Thanks for the suggestions btw I will look into the AppleScript approach to see if I have any luck getting it to work 👍🏻
